core/src/main/java/org/mule/runtime/core/message/ErrorBuilder.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.message;
+
+import static org.mule.runtime.core.config.ExceptionHelper.getRootMuleException;
+import static org.mule.runtime.core.message.ErrorTypeBuilder.GENERAL;
+import static org.mule.runtime.core.util.Preconditions.checkState;
+
+import org.mule.runtime.api.message.Error;
+import org.mule.runtime.api.message.ErrorType;
+import org.mule.runtime.api.message.MuleMessage;
+import org.mule.runtime.core.api.MuleException;
+
+/**
+ * Builder for {@link Error} instances.
+ *
+ * @since 4.0
+ */
+public final class ErrorBuilder {
+
+  private Throwable exception;
+  private String description;
+  private String detailedDescription;
+  private ErrorType errorType;
+  private MuleMessage errorMessage;
+
+  public static ErrorBuilder builder() {
+    return new ErrorBuilder();
```

Comment:
Does an error with no throwable or message make any sense?  If not we shouldn't expose a builder() IMO